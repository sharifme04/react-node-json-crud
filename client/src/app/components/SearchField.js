import React from 'react'
import PropTypes from 'prop-types'

const SearchField = ({value, handleChange}) => {
  return (
    <div className="search-field">
    <input
      type="text"
      className="form-control"
      placeholder="search by title"
      value={value}
      onChange={handleChange}
    />
  </div>
  )
}

SearchField.propTypes = {
    value: PropTypes.any,
    handleChange: PropTypes.func
}

export default SearchField