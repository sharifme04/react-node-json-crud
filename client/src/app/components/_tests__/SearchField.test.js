import React from "react";
import renderer from "react-test-renderer";
import SearchField from "../SearchField";

test("snapshot SearchField renders with empty props", () => {
  const component = renderer.create(<SearchField />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot SearchField renders with props", () => {
  const component = renderer.create(
    <SearchField
      value={'value'}
      handleChange={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
