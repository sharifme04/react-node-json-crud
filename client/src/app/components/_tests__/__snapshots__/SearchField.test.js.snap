// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SearchField renders with empty props 1`] = `
<div
  className="search-field"
>
  <input
    className="form-control"
    placeholder="search by title"
    type="text"
  />
</div>
`;

exports[`snapshot SearchField renders with props 1`] = `
<div
  className="search-field"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="search by title"
    type="text"
    value="value"
  />
</div>
`;
